config/config.go

```diff
@@ -423,6 +424,16 @@ type PyroscopeConfiguration struct {
 	ProfileID       string `json:"profile_id" envconfig:"CONVOY_PYROSCOPE_PROFILE_ID"`
 }
 
+type HCPVaultConfig struct {
+	ClientID      string        `json:"client_id" envconfig:"HCP_CLIENT_ID"`
+	ClientSecret  string        `json:"client_secret" envconfig:"HCP_CLIENT_SECRET"`
+	OrgID         string        `json:"org_id" envconfig:"HCP_ORG_ID"`
+	ProjectID     string        `json:"project_id" envconfig:"HCP_PROJECT_ID"`
+	AppName       string        `json:"app_name" envconfig:"HCP_APP_NAME"`
+	SecretName    string        `json:"secret_name" envconfig:"HCP_SECRET_NAME"`
+	CacheDuration time.Duration `json:"cache_duration" envconfig:"HCP_CACHE_DURATION"`
```

Comment:
I think we need to prefix the env vars with `CONVOY_` to be consistent with the rest. I know we set a prefix but I'm not sure if it's automatically applied.